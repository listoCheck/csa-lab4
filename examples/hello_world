S" Hello, world!" VARIABLE hello_world
0x0 VARIABLE index
0x0 VARIABLE len
_start:
    lit len
    lit hello_world @
    !
    lit index
    lit hello_world
    lit 4
    +
    !
while:
    lit index @ @
    out 2
    lit index
    dup
    @
    lit 4
    +
    !
    lit 1
    lit len @
    -
    dup
    if end
    lit len
    swap
    !
    jump while
end:
    halt
