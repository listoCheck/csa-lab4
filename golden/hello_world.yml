in_code: |-
  _start:
    lit 72
    out 2

    lit 101
    out 2

    lit 108
    out 2

    lit 108
    out 2

    lit 111
    out 2

    lit 32
    out 2

    lit 87
    out 2

    lit 111
    out 2

    lit 114
    out 2

    lit 108
    out 2

    lit 100
    out 2

    lit 33
    out 2

    halt

in_stdin: |-
  foo
out_code_hex: |-
  0 - 1548 - lit 72
  1 - 1602 - out 2
  2 - 1565 - lit 101
  3 - 1602 - out 2
  4 - 156C - lit 108
  5 - 1602 - out 2
  6 - 156C - lit 108
  7 - 1602 - out 2
  8 - 156F - lit 111
  9 - 1602 - out 2
  10 - 1520 - lit 32
  11 - 1602 - out 2
  12 - 1557 - lit 87
  13 - 1602 - out 2
  14 - 156F - lit 111
  15 - 1602 - out 2
  16 - 1572 - lit 114
  17 - 1602 - out 2
  18 - 156C - lit 108
  19 - 1602 - out 2
  20 - 1564 - lit 100
  21 - 1602 - out 2
  22 - 1521 - lit 33
  23 - 1602 - out 2
  24 - 14 - halt
out_stdout: |2

  Машинный код
  0 - 1548 - lit 72
  1 - 1602 - out 2
  2 - 1565 - lit 101
  3 - 1602 - out 2
  4 - 156C - lit 108
  5 - 1602 - out 2
  6 - 156C - lit 108
  7 - 1602 - out 2
  8 - 156F - lit 111
  9 - 1602 - out 2
  10 - 1520 - lit 32
  11 - 1602 - out 2
  12 - 1557 - lit 87
  13 - 1602 - out 2
  14 - 156F - lit 111
  15 - 1602 - out 2
  16 - 1572 - lit 114
  17 - 1602 - out 2
  18 - 156C - lit 108
  19 - 1602 - out 2
  20 - 1564 - lit 100
  21 - 1602 - out 2
  22 - 1521 - lit 33
  23 - 1602 - out 2
  24 - 14 - halt

  source LoC: 38 code instr: 25
  ============================================================
  [tick 0] LIT - подготовка
  [tick 1] LIT - загрузка значения 72
  [tick 1] TOS -> STACK
  [tick 2] EMIT
  Output char 'H'
  [tick 3] LIT - подготовка
  [tick 4] LIT - загрузка значения 101
  [tick 4] TOS -> STACK
  [tick 5] EMIT
  Output char 'e'
  [tick 6] LIT - подготовка
  [tick 7] LIT - загрузка значения 108
  [tick 7] TOS -> STACK
  [tick 8] EMIT
  Output char 'l'
  [tick 9] LIT - подготовка
  [tick 10] LIT - загрузка значения 108
  [tick 10] TOS -> STACK
  [tick 11] EMIT
  Output char 'l'
  [tick 12] LIT - подготовка
  [tick 13] LIT - загрузка значения 111
  [tick 13] TOS -> STACK
  [tick 14] EMIT
  Output char 'o'
  [tick 15] LIT - подготовка
  [tick 16] LIT - загрузка значения 32
  [tick 16] TOS -> STACK
  [tick 17] EMIT
  Output char ' '
  [tick 18] LIT - подготовка
  [tick 19] LIT - загрузка значения 87
  [tick 19] TOS -> STACK
  [tick 20] EMIT
  Output char 'W'
  [tick 21] LIT - подготовка
  [tick 22] LIT - загрузка значения 111
  [tick 22] TOS -> STACK
  [tick 23] EMIT
  Output char 'o'
  [tick 24] LIT - подготовка
  [tick 25] LIT - загрузка значения 114
  [tick 25] TOS -> STACK
  [tick 26] EMIT
  Output char 'r'
  [tick 27] LIT - подготовка
  [tick 28] LIT - загрузка значения 108
  [tick 28] TOS -> STACK
  [tick 29] EMIT
  Output char 'l'
  [tick 30] LIT - подготовка
  [tick 31] LIT - загрузка значения 100
  [tick 31] TOS -> STACK
  [tick 32] EMIT
  Output char 'd'
  [tick 33] LIT - подготовка
  [tick 34] LIT - загрузка значения 33
  [tick 34] TOS -> STACK
  [tick 35] EMIT
  Output char '!'
  [tick 36] HALT

out_log: |-
  DEBUG    root:machine.py:500 Tick: 0, PC: 0, Stack: [0, 0], Input: ['f', 'o', 'o'], Output: [], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 2, PC: 1, Stack: [72, 0], Input: ['f', 'o', 'o'], Output: [], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 3, PC: 2, Stack: [0, 0], Input: ['f', 'o', 'o'], Output: ['H'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 5, PC: 3, Stack: [101, 0], Input: ['f', 'o', 'o'], Output: ['H'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 6, PC: 4, Stack: [0, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 8, PC: 5, Stack: [108, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 9, PC: 6, Stack: [0, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 11, PC: 7, Stack: [108, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 12, PC: 8, Stack: [0, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 14, PC: 9, Stack: [111, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 15, PC: 10, Stack: [0, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l', 'o'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 17, PC: 11, Stack: [32, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l', 'o'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 18, PC: 12, Stack: [0, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l', 'o', ' '], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 20, PC: 13, Stack: [87, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l', 'o', ' '], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 21, PC: 14, Stack: [0, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l', 'o', ' ', 'W'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 23, PC: 15, Stack: [111, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l', 'o', ' ', 'W'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 24, PC: 16, Stack: [0, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l', 'o', ' ', 'W', 'o'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 26, PC: 17, Stack: [114, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l', 'o', ' ', 'W', 'o'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 27, PC: 18, Stack: [0, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 29, PC: 19, Stack: [108, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 30, PC: 20, Stack: [0, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 32, PC: 21, Stack: [100, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 33, PC: 22, Stack: [0, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l', 'd'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 35, PC: 23, Stack: [33, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l', 'd'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 36, PC: 24, Stack: [0, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l', 'd', '!'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  DEBUG    root:machine.py:504 Tick: 37, PC: 24, Stack: [0, 0], Input: ['f', 'o', 'o'], Output: ['H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l', 'd', '!'], tos: 0, a: 0, b: [], dump: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
  EOF
